thomaspoignant/go-feature-flag
# feat: Add config to allow surpressing notification on launch (flag cache load)

## Description

This pull request introduces a new feature to disable notifications on initialization and refactors the caching mechanism to support this feature. It also includes updates to the test suite to ensure the new functionality works as expected.

Problem
* go-feature-flag always sent notification of diff to all Notifierson ffClient initialization because it detects the difference between the initial cache (= always empty) and the flags retrieved.

Resolution
* introduce `DisableNotificationOnInit` configuration key. If set to true, avoid sending the notification on cache initialization (it will still send notification when the change is detected AFTER initialization)
  * Since the behavior (should) stays the same if the new config is NOT set, it should not introduce any breaking change.

How to test
* I've added some tests to make sure the behavior stays the same when the new config is not explicitly set. You can run `go test ./...`
* Or, simply initialize `ffClient` should do the trick too.

## Closes issue(s)

Resolve #2532 

## Checklist
